extensions/wikia/Email/MobileApplicationsLinksGenerator.class.php

```diff
@@ -0,0 +1,194 @@
+		'en' => 'us',
+		'zh' => 'cn',
+		'zh-tw' => 'tw',
+		'fr' => 'fr',
+		'de' => 'de',
+		'it' => 'it',
+		'ja' => 'jp',
+		'pl' => 'pl',
+		'pt' => 'pt',
+		'pt-br' => 'br',
+		'ru' => 'ru',
+		'es' => 'es',
+	];
+
+	private $language;
+
+	/**
+	 * MobileApplicationsLinksGenerator constructor.
+	 *
+	 * @param string $language - language for localized store
+	 */
+	public function __construct( $language ) {
+		$this->language = $language;
+	}
+
+	/**
+	 * Generates links for mobile applications for current wiki.
+	 *
+	 * @return array - currently will contains 2 elements, link for app store and google play
+	 */
+	public function generate() {
+		global $wgMemc;
+
+		$mobileApplicationsLinks =
+			$wgMemc->get( $this->createCacheKeyForMobileApplicationsLinks() );
+
+		if ( is_bool( $mobileApplicationsLinks ) ) {
+			$mobileApplicationsLinks = [];
+			$response = $this->fetchMobileApplicationsDetails();
+
+			if ( $response && $this->applicationsExistIn( $response ) ) {
+				$mobileApplications = json_decode( $response, true );
+				$mobileApplicationsLinks = $this->traverseThrough( $mobileApplications );
+			}
+
+			$mobileApplicationsLinks =
+				$this->fillWithWikiaAccountLinksIfEmpty( $mobileApplicationsLinks );
+
+			// Even if response is not valid (for example there are some difficulties with mobile applications service)
+			// result will be cached to prevent too long delays in email generation
+			$wgMemc->set( $this->createCacheKeyForMobileApplicationsLinks(),
+				$mobileApplicationsLinks, time() + self::MOBILE_APPLICATIONS_LINKS_EVICTION_TIME );
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	/**
+	 * Fills array with wikia mobile application store link if application was not found
+	 *
+	 * @param array $mobileApplicationsLinks
+	 * @return array with links to app store and google play
+	 */
+	private function fillWithWikiaAccountLinksIfEmpty( $mobileApplicationsLinks ) {
+		if ( empty ( $mobileApplicationsLinks[self::ANDROID_PLATFORM] ) ) {
+			$mobileApplicationsLinks[self::ANDROID_PLATFORM] =
+				$this->generateUrlForWikiaGooglePlay();
+		}
+
+		if ( empty( $mobileApplicationsLinks[self::IOS_PLATFORM] ) ) {
+			$mobileApplicationsLinks[self::IOS_PLATFORM] = $this->generateUrlForWikiaAppStore();
+		}
+
+		return $mobileApplicationsLinks;
+	}
+
+	private function generateUrlForWikiaGooglePlay() {
+		return sprintf( self::WIKIA_GOOGLE_PLAY_URL, $this->prepareLanguageForGooglePlay() );
+	}
+
+	private function prepareLanguageForGooglePlay() {
+		return $this->language;
+	}
+
+	private function generateUrlForWikiaAppStore() {
+		return sprintf( self::WIKIA_APP_STORE_URL, $this->prepareLanguageForAppStore() );
+	}
+
+	private function prepareLanguageForAppStore() {
+		return empty ( self::$appStoreLanguagesMapping[$this->language] ) ? 'us'
+			: self::$appStoreLanguagesMapping[$this->language];
+	}
+
+	private function createCacheKeyForMobileApplicationsLinks() {
+		return wfMemcKey( 'mobileApplicationsLinks', $this->language );
```

Comment:
Actually this language is not a wiki language, but target user language. I've changed code to store only links with `%s` in language place, to store less links in cache. `%s` is then replaced with target user language.